### ISSUE TITLE
Terminal is mouse wheel is scrolling too much

### ISSUE THREAD

**Issue by Tyriar** [xtermjs/xterm.js#1805]
Original issue: https://github.com/Microsoft/vscode/issues/63669

Using these mouse settings on Windows 10:

![image](https://user-images.githubusercontent.com/2193314/49312226-57cc1900-f498-11e8-9dd4-7c9e2cf8460d.png)

The terminal seems to scroll 7 lines instead of 3.

Code pointer:

https://github.com/xtermjs/xterm.js/blob/290c95ad71f8b1264139a7ed6f070c4741a8987e/src/Viewport.ts#L160-L190

**Comment by skprabhanjan** [comment]
@Tyriar , Can i take a look at this ?

**Comment by Tyriar** [comment]
@skprabhanjan sure 😃

**Comment by skprabhanjan** [comment]
@Tyriar .
Firstly I am not able to reproduce this exactly as I am using Mac, But still i checked the code for any possible thing that might be causing this issue and I found none.
So could you please help me proceed further with this ?
Thanks :)

**Comment by Tyriar** [comment]
@skprabhanjan if you don't have a Windows machine+mouse I don't think you could help on this one unfortunately.

**Comment by skprabhanjan** [comment]
@Tyriar , I get it :)
I will leave this fix for someone else then, thanks :)

**Comment by thenoobtester** [comment]
I have the exact same issue . Windows is set to 3 lines but terminal jumps 7 (real cmd is consistent with global settings). I changed to 2 and it jumps 4, 1 and it jumps 2. It's really annoying. It drove me nuts until I figured out it wasn't a real terminal and that xterm.js was the culprit. Hope it will be fixed soon.

**Comment by Tyriar** [comment]
> until I figured out it wasn't a real terminal and that xterm.js was the culprit

💔

**Comment by thenoobtester** [comment]
Sorry about that I didn't mean it in a bad way :)

**Comment by Tyriar** [comment]
❤️

**Comment by DanJBower** [comment]
Just wanted to add to this. I'm still having the same issue, but it appears to be dependant on monitor scale in the windows settings. I think this might be how VSCode interprets a scroll. Today I had my external monitor plugged in, a 24" 1920 x 1080 screen with 100% scale in the display settings. On that monitor, the terminal works as I would expect and so does the scroll in the main editor window of VSCode. However, on my laptop monitor, a 4k 15" screen which I have at 250% scale, the terminal scrolls 14 lines and the main editor window about 6 / 7. So I'm not convinced this is an issue with xterm but possibly more an issue with VSCode

**Comment by Tyriar** [comment]
These settings on Windows (3 lines each time) appear to translate in Chromium to deltaMode `WheelEvent.DOM_DELTA_PIXEL` and 100 pixels

![image](https://user-images.githubusercontent.com/2193314/60673748-551a1380-9e2d-11e9-9547-8fcebf7de038.png)

This does end up scrolling 100 pixels which ends up being 6 rows. Zooming to 200% will still scroll 6 rows.

Changing it to 1 line in Windows settings makes each wheel event 33.33 pixels and 2 rows

So it seems like it's working as intended so far, just the event from Chromium isn't matching what I'm expecting.

---

Switching to one screen at a time scrolls exactly one screen.

**Comment by Tyriar** [comment]
@eamodio and I went back and forth on this quite a bit. Chromium by default uses 100px for 3 lines on Windows (which might not actually be 3 lines depending on the content), Chromium Edge does something different, as does Firefox. Trackpads can also act a little differently.

We decided to just go with adding a setting to allow people to configure it manually if they have issues as browsers don't (and cannot) respect the Windows setting (#2485). This seems preferable to fighting the browser with browser-specific hacks that will probably fall apart over time.
